File: backend/app/models/brand.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Index, String, Text, func
from sqlalchemy.dialects.postgresql import ARRAY, JSONB, UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base
from app.models.base import TimestampMixin, UUIDPrimaryKeyMixin


class BrandContext(UUIDPrimaryKeyMixin, TimestampMixin, Base):
    __tablename__ = "brand_contexts"
    __table_args__ = (
        Index("idx_brand_contexts_user_id", "user_id"),
    )

    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    website_url: Mapped[str | None] = mapped_column(Text)

    primary_colors: Mapped[list] = mapped_column(ARRAY(String), default=list)
    secondary_colors: Mapped[list] = mapped_column(ARRAY(String), default=list)
    accent_colors: Mapped[list] = mapped_column(ARRAY(String), default=list)

    primary_font: Mapped[str | None] = mapped_column(String(100))
    secondary_font: Mapped[str | None] = mapped_column(String(100))

    style_keywords: Mapped[list] = mapped_column(ARRAY(String), default=list)
    mood: Mapped[str | None] = mapped_column(String(100))
    industry: Mapped[str | None] = mapped_column(String(100))

    logo_urls: Mapped[list] = mapped_column(ARRAY(String), default=list)
    reference_image_urls: Mapped[list] = mapped_column(ARRAY(String), default=list)

    visual_patterns: Mapped[dict] = mapped_column(JSONB, default=dict)
    status: Mapped[str] = mapped_column(String(50), default="active")
    last_scraped_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    metadata_: Mapped[dict] = mapped_column("metadata", JSONB, default=dict)

File: frontend/src/types/brand.ts
export interface Brand {
  id: string;
  user_id: string;
  name: string;
  website_url: string | null;
  primary_colors: string[];
  secondary_colors: string[];
  accent_colors: string[];
  primary_font: string | null;
  secondary_font: string | null;
  style_keywords: string[];
  mood: string | null;
  industry: string | null;
  logo_urls: string[];
  reference_image_urls: string[];
  visual_patterns: Record<string, unknown>;
  status: string;
  last_scraped_at: string | null;
  metadata: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface BrandCreateInput {
  name: string;
  website_url?: string;
  primary_colors?: string[];
  secondary_colors?: string[];
  accent_colors?: string[];
  primary_font?: string;
  secondary_font?: string;
  style_keywords?: string[];
  mood?: string;
  industry?: string;
}

export interface BrandUpdateInput {
  name?: string;
  website_url?: string;
  primary_colors?: string[];
  secondary_colors?: string[];
  accent_colors?: string[];
  primary_font?: string;
  secondary_font?: string;
  style_keywords?: string[];
  mood?: string;
  industry?: string;
}

export interface WebsiteAnalysis {
  colors: string[];
  typography: { primary: string; secondary: string };
  style: { dominant: string; mood: string };
  logos: string[];
}
